plenum/server/node.py

```diff
@@ -3178,7 +3175,11 @@ def executeBatch(self, view_no, pp_seq_no: int, pp_time: float,
                                              ledger_id,
                                              view_no, pp_seq_no)
 
-        batch_committed_msg = BatchCommitted([req.as_dict for req in valid_reqs],
+        reqs = []
+        for req_key in valid_reqs_keys:
+            if req_key in self.requests:
```

Comment:
I think we should create and send `BatchCommitted` only if all requests exist. If there is at least one req which is not in self.requests - then log a warning and do not create and send BatchCommitted at all.

Reply:
Yes, sure.